File: html_tags/open_tag.py
# 열기만 하는 태그
def open_tag(tag_name, types=[], values=[]):
    # 시작 태그 작성
    tag_start = f"<{tag_name}"

    # types를 토대로 클로저 함수 생성
    def type_plus(tag_start, types, values):
        # 시작 태그 지정
        tag = tag_start
        # 만약, values가 리스트면서 값이 없다면
        # tag에 type들 모두 대입하면서
        # values 적용
        if type(values) == list and values != []:
            for index, type_name in enumerate(types):
                tag += f" {type_name}={values[index]}"
        # values가 리스트면서 값이 있다면
        # values가 빈배열이 맞다면
        elif type(values) == list and values == []:
            for type_name in types:
                tag += f" {type_name}"
        # 모두 더하면 닫기
        tag += ">"
        # tag 반환
        return tag

    # type_plus 실행
    tag = type_plus(tag_start, types, values)
    # tag return
    return tag

File: html_tags/openTag.js
// 열기만 하는 코드 제작
// 클로저 함수
// DOCTYPE 선언 같은 경우 사용됨

// 첫번째 파라미터 : 태그 네임 tagName
// 두번째 파라미터 : 속성명 typeps(배열형식)
// 세번째 파라미터 : 속성값 values(배열형식)

const openTag = (tagName, types = [], values = []) => {
  // 태그
  let tag = `<${tagName}`;

  // 클로저함수
  const typePlus = (types, values) => {
    // 속성들을 분해함
    types.forEach((element, index) => {
      //만약에 types가 배열 형식이면서
      // 빈 배열이 아닐 때
      if (typeof values === "object" && values.length > 0) {
        // 속성명에 =를 붙이면서
        // 인덱스에 맞는 value값 추가
        tag += ` ${element}=${values[index]}`;
      }
      //만약에 types가 배열 형식이면서
      // 빈 배열이 아닐 때
      else if (typeof values === "object" && values.length <= 0) {
        // 속성명만 추가
        tag += ` ${element}`;
      }
    });

    // 태그 끝
    tag += `>`;
  };

  //   typePlus 함수 실행
  typePlus(types, values);

  //   태그 return
  return tag;
};


// 테스트
const testTag1 = openTag("div", ["class", "id"], ['"container"','"main"']);
const testTag2 = openTag("div");

// 출력
console.log(testTag1);
console.log(testTag2);